### PR TITLE
Honor optional allFiles on make_non_static

### LINKED ISSUE #321: Honor make_non_static allFiles
## Leftover

Honor optional **`allFiles`** (and optional **`sourceFile` when `allFiles` is true**) on the existing `make_non_static` tool so it matches document-rewrite AllFiles siblings (`make_static`, `convert_to_async`, `add_null_checks`, `convert_property`, `move_type_to_namespace`).

### Observed

- Master `fddc601a93c6d988cb03058111de50b30f204309` after #320 (`Honor make_static allFiles`). Build and Test https://github.com/JoshuaRamirez/RoslynMcpServer/actions/runs/33632125197 SUCCESS (ubuntu + windows + macos); Code Quality https://github.com/JoshuaRamirez/RoslynMcpServer/actions/runs/33632125208 SUCCESS (quality + Format Check).
- Paid AllFiles series (do not reopen): organize-trio, format_document, simplify_name, add_braces, remove_braces, convert_expression_body, convert_to_interpolated_string, convert_foreach_linq, convert_to_pattern_matching, invert_if, convert_property, convert_to_async, rename_file_to_match_type, add_null_checks, encapsulate_field, use_base_type, move_type_to_file, move_type_to_namespace, make_static (#319/#320).
- No open leftover. No product PRs (Dependabot only — leave them).
- **`MakeNonStaticParams` is the next honest holdout:** required `string SourceFile`, required `int StartLine` / `StartColumn` / `EndLine` / `EndColumn`, optional `SymbolName` / `Preview`, **no `AllFiles`**. `Validate` always requires existing absolute `.cs` sourceFile + valid selection range. `ExecuteCoreAsync` resolves **one** ordinary static method from the selection, rejects already-instance / extension / virtual / override / abstract / extern / static-class / interface / interface-implementing methods, then removes `static` and rewrites type-name invocations and method-group conversions to a valid instance receiver (or `this` in the same type). Call sites without a valid instance receiver or with unsafe null-conditional access (`?.`) throw (`NoValidInstanceReceiver` / `InvalidSelection`).
- Collect path is paid: selection → `IMethodSymbol` → `ValidateMethodCanBeMadeNonStatic` → `BuildPlanAsync` / `FindCallSiteEditsAsync` / `ResolveCallSiteRewrite` → rewrite. Clear **skip-not-throw** paths for bulk ineligible methods (already instance, extension, static class, no receiver, conditional access, uneditable).
- Tool schema requires `solutionPath`, `sourceFile`, `startLine`, `startColumn`, `endLine`, `endColumn`; `additionalProperties = false`. CLI/README omit allFiles.
- Do **not** invent `convert_to_block_body` allFiles (`convert_expression_body` `Direction=ToBlockBody` already has AllFiles). Do not reopen paid AllFiles series. Do not invent `rename_namespace` allFiles in this leftover (separate track if ever). Do not file a second leftover.

### In scope (bounded)

- Add `bool AllFiles { get; init; }` (default false) to `MakeNonStaticParams`. Make `SourceFile` `string?`. Make `StartLine` / `StartColumn` / `EndLine` / `EndColumn` `int?` (required only when `allFiles` is false). Keep `SymbolName` / `Preview` with today's single-site semantics.
- Validation: when `allFiles` is false, keep today's sourceFile + selection required / absolute / `.cs` / exists and range bounds. When `allFiles` is true, skip sourceFile/selection validation; still allow optional `sourceFile` to limit the walk to that one file (same optional-sourceFile pattern as `make_static` / `convert_to_async` / other AllFiles siblings); reject `startLine`, `startColumn`, `endLine`, `endColumn`, or `symbolName` (`MissingRequiredParam`) as incompatible with bulk mode. `preview` remains valid with allFiles.
- Execute: `allFiles: false` (default) keeps today's single-site make-non-static (including throws). `allFiles: true` walks every C# document (`FilePath != null && EndsWith(".cs")`, or the optional single `sourceFile`). Per file, consider every **ordinary** `MethodDeclarationSyntax` (same kind filter as `ValidateMethodCanBeMadeNonStatic`). For each method that can be made an instance method (already static; not extension; not virtual/override/abstract/extern; not in a static class; not interface / interface-implementing; a valid instance receiver exists for every call site that needs rewrite; document editable), apply today's rewrite (remove `static`, update type-name call sites and method-group conversions to that receiver or `this`). **Skip** ineligible methods rather than throwing (already-instance, extension, static-class, no-valid-receiver, unsafe `?.`, uneditable). Aggregate `FilesModified` / preview. Process documents and methods in a deterministic order. If a later rewrite conflicts with an earlier one, skip the later claim rather than throwing mid-walk. If every method is a no-op, succeed with empty changes — do not throw single-site `SymbolNotFound` / `AlreadyInstance` / `InvalidSymbolKind` / `InvalidSelection` / no-valid-receiver mid-walk for skips.
- Preview: `preview: true` with `allFiles: true` returns aggregated `PendingChanges` and writes nothing.
- Wire Contracts → Core (`MakeNonStaticOperation` Validate + Execute / ExecuteAllFilesAsync) → Server tool schema/args (`allFiles` boolean default false; `sourceFile` not required; selection ints not required when allFiles; required `solutionPath`; `additionalProperties = false`) → CLI help → README tools table → CHANGELOG Unreleased.
- Tests: omitted/`false` keeps today's single-site behavior; `allFiles: true` makes eligible already-static methods instance and skips already-instance / extension / virtual / interface / static-class / no-receiver / ineligible; `allFiles: true` without sourceFile/selection succeeds; `allFiles: false` without them still `MissingRequiredParam`; `allFiles: true` with startLine/symbolName is `MissingRequiredParam`; preview writes nothing; schema accepts `allFiles`. Prefer Core operation tests plus tool schema coverage. Mirror `make_static` AllFiles PR #320 patterns where applicable. Inverse eligibility of `make_static` (already-static methods that can become instance, skip ineligible).

### Out of scope

- No new tool. No Dependabot. No second leftover. Do not invent unrelated features. Do not invent `convert_to_block_body` allFiles. Do not invent `rename_namespace` allFiles in this PR. Do not reopen paid AllFiles series. No Bank of America STAR. Do not change single-site selection/error codes or the instance-receiver eligibility rules.

### Done when

PR opens against master from `fddc601a93c6d988cb03058111de50b30f204309`, CI green, Copilot/Codex threads resolved, squash-merged, this issue closed.

Closes pattern for the implementing PR: `Closes #<this-issue>`.

### Spec links

- Siblings: `make_static` AllFiles (#320) / `convert_to_async` / `add_null_checks` / `convert_property` / `move_type_to_namespace` AllFiles
- Current single-site: `MakeNonStaticOperation.Validate` / `ValidateMethodCanBeMadeNonStatic` / `FindCallSiteEditsAsync` / `ResolveCallSiteRewrite`
- Tool: make_non_static / CLI
- Base SHA: `fddc601a93c6d988cb03058111de50b30f204309`

### PR DESCRIPTION
**Pull request by JoshuaRamirez**
## Description

Honor optional `allFiles` (and optional `sourceFile` when `allFiles` is true) on the existing `make_non_static` tool so it matches document-rewrite AllFiles siblings (`make_static`, `convert_to_async`, `add_null_checks`, `convert_property`, `move_type_to_namespace`).

Omitted / `false` keeps today's single-site make-non-static (`sourceFile` + selection required, including today's throws). `allFiles: true` walks every C# document (`FilePath` ends with `.cs`, or the optional single `sourceFile`) and makes every eligible ordinary static method an instance method using today's rewrite (remove `static`, update type-name call sites / method-group conversions to a valid instance receiver or `this`). Already-instance, extension, virtual/override/abstract/extern, static-class, interface / interface-implementing, no-valid-receiver, uneditable, conditional-access, and otherwise ineligible methods are skipped rather than failing the walk. If every method is a no-op, succeeds with empty changes. `preview: true` with `allFiles: true` returns aggregated `PendingChanges` and writes nothing.

Preview descriptions count conversions per declaration document (not only the scan file), so a split partial gets "Make method an instance method" on both declaration files.

Does not invent `convert_to_block_body` allFiles or `rename_namespace` allFiles. Does not reopen the paid AllFiles series. Leaves Dependabot alone. No second leftover.

## Related Issues

Closes #321

## Type of Change

- [x] ✨ New feature (non-breaking change that adds functionality)
- [x] ✅ Test additions or updates
- [x] 📚 Documentation update

## Testing

- [x] New tests added for new functionality
- [x] All existing tests pass (locally, targeted suites)

**Test Configuration**:
- OS: Linux
- .NET Version: 9.0.317

Local results:
- `MakeNonStaticOperationTests`: 47 passed
- `MakeNonStaticToolTests`: 11 passed
- `RoslynMcp.Cli.Tests`: 158 passed
- `dotnet format --verify-no-changes`: passed
- `dotnet build -c Release /p:TreatWarningsAsErrors=true`: 0 warnings, 0 errors

Core operation tests cover omitted/`false` single-site, `allFiles: true` eligible/skip, missing sourceFile/selection, incompatible `startLine`/`symbolName`, optional `sourceFile` walk limit, preview writes nothing, empty-change success, partial-method both-declaration rewrite, and preview descriptions on both partial declaration files. Server schema tests accept `allFiles` (default false; `sourceFile` / selection not required). CLI help lists `--all-files` as optional.

## Checklist

- [x] My code follows the project's code style guidelines
- [x] I have performed a self-review of my code
- [x] I have added XML documentation for public APIs
- [x] I have made corresponding changes to the documentation (README tools table, CHANGELOG Unreleased, CLI help)
- [x] I have added tests that prove my feature works
- [x] New and existing unit tests pass locally with my changes

<div><a href="https://cursor.com/agents/bc-4a5b0d7f-381b-4cb4-a0dd-134da018671b?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-4a5b0d7f-381b-4cb4-a0dd-134da018671b&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs [hunk 1]
```diff
@@ -142,18 +176,369 @@ protected override async Task<RefactoringResult> ExecuteCoreAsync(
...
+
+                    var declarationDocuments = await GetDeclarationDocumentsAsync(canonical, cancellationToken);
+                    if (declarationDocuments.Any(declarationDocument =>
+                            !IsDocumentEditable(declarationDocument, Context.Workspace)))
+                    {
+                        continue;
+                    }
+
+                    var plan = await BuildPlanAsync(canonical, cancellationToken);
+                    if (PlanConflictsWithClaimedSpans(plan, claimedSpans))
```
[the remark this answers is not in the document]
Comment (JoshuaRamirez) [reply src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs:254]: SKIP — out of leftover #321 scope.

Leftover #321 is honor optional `allFiles` on `make_non_static` matching document-rewrite siblings (`make_static`). `ExecuteAllFilesAsync` already matches `MakeStaticOperation`: independent `BuildPlanAsync` against `Context.Solution`, then one aggregated `ApplyPlanAsync`. Combined post-aggregation rebind/compile-check is extra robustness, not leftover-pinned (skip already-instance/extension/virtual/static-class/interface/no-receiver/unsafe `?.`; bulk skip-not-throw; inverse eligibility). `WouldRebindToHidingMember` already runs per plan. Requiring a second global revalidation is a new feature, not leftover-scope correctness.

File: src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs [hunk 2]
```diff
@@ -142,18 +176,369 @@ protected override async Task<RefactoringResult> ExecuteCoreAsync(
...
+        catch (Exception ex) when (ex is ArgumentException or NotSupportedException or PathTooLongException)
+        {
+            wanted = sourceFile;
+        }
+
+        return documents
+            .Where(d => string.Equals(
+                PathResolver.NormalizePath(d.FilePath!),
+                wanted,
+                StringComparison.OrdinalIgnoreCase))
```
[the remark this answers is not in the document]
Comment (JoshuaRamirez) [reply src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs:492]: SKIP — out of leftover #321 scope / sibling-pin.

`FilterDocumentsBySourceFile` uses `PathResolver.NormalizePath` + `StringComparison.OrdinalIgnoreCase`, identical to `MakeStaticOperation.FilterDocumentsBySourceFile`. Tests include `MakeNonStatic_AllFilesTrue_OptionalSourceFile_MatchesIgnoreCase`. Preferring an exact match first for `Foo.cs` vs `foo.cs` is an edge-case new feature, not leftover-pinned.

File: src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs [hunk 3]
```diff
@@ -142,18 +176,369 @@ protected override async Task<RefactoringResult> ExecuteCoreAsync(
...
+
+                    foreach (var declaration in plan.Declarations)
+                        claimedSpans.Add((declaration.SyntaxTree, declaration.Span));
+                    foreach (var callSite in plan.CallSites)
+                        claimedSpans.Add((callSite.SyntaxTree, callSite.Span));
+
+                    acceptedDeclarations.AddRange(plan.Declarations);
+                    acceptedCallSites.AddRange(plan.CallSites);
+                    convertedCountByDoc[document.Id] =
+                        convertedCountByDoc.GetValueOrDefault(document.Id) + 1;
```
[the remark this answers is not in the document]
Comment (JoshuaRamirez) [reply src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs]: TAKE — leftover-scope correctness: partial-method both-declaration + allFiles preview descriptions.

`ExecuteAllFilesAsync` increments `convertedCountByDoc[document.Id]` for the scan document (`MakeNonStaticOperation.cs` ~264–265), then preview uses that count for per-file `PendingChange` descriptions (~320–327). Partial methods rewrite both declaration documents (`GetPartialMethodParts` / test `MakeNonStatic_AllFilesTrue_CalledPartialMethod_MakesBothDeclarationsInstance`), but the other declaration file would get "Update call sites of methods made instance" instead of "Make method an instance method".

Ask: increment `convertedCountByDoc` for each declaration document in the accepted plan (via `GetDeclarationDocumentsAsync` / `plan.Declarations`), not only the currently scanned document. Add a preview `allFiles` test with a split partial so both declaration files get the converted-method description. Do not invent `convert_to_block_body` / `rename_namespace` allFiles.

File: src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs [hunk 4]
```diff
@@ -142,18 +176,369 @@ protected override async Task<RefactoringResult> ExecuteCoreAsync(
...
+    /// <c>AddNullChecksOperation.ExecuteAllFilesAsync</c> /
+    /// <c>ConvertPropertyOperation.ExecuteAllFilesAsync</c> /
+    /// <c>MakeStaticOperation.ExecuteAllFilesAsync</c>) and makes every
+    /// eligible ordinary static method an instance method using today's
+    /// rewrite (remove <c>static</c>, update type-name call sites and
+    /// method-group conversions to a valid instance receiver or
+    /// <c>this</c>). Optional <c>sourceFile</c> limits the walk to that
+    /// one file. Already-instance, extension, virtual/override/abstract/extern,
+    /// static-class, interface / interface-implementing, no-valid-receiver,
+    /// uneditable, conditional-access, and otherwise ineligible methods are
```
[the remark this answers is not in the document]
Comment (JoshuaRamirez) [reply src/RoslynMcp.Core/Refactoring/Extract/MakeNonStaticOperation.cs:192]: SKIP — docs nit; leftover behavior already matches inverse eligibility.

Leftover #321 pins skip unsafe `?.`, inverse of `make_static`. Implementation already skips only when `BuildPlanAsync` throws (receiver rewrite would be unsafe). Test `MakeNonStatic_AllFilesTrue_SkipsConditionalAccess` documents that `other?.Log()` stays and `Log` is still eligible. The XML summary listing "conditional-access" next to skip-not-throw is a wording over-summarize, not leftover-scope runtime correctness.
